### DISCUSSION TITLE
Unable to dynamically set deploy environment using inputs in a GitHub workflow

### DISCUSSION THREAD

**Discussion by lktslionel** [community/community#26697] [Actions]
<p>Hello,</p>
<p>I have the following GitHub deployment workflow:</p>
<pre><code class="lang-auto">name: Deployment Workflow 
on: 
  workflow_dispatch:
    inputs:
      component-version:
        description: 'Component version you want to deploy'
        required: true
        type: string

      env-id:
        description: 'Environment ID'
        type: environment
        required: true 

env: 
  COMPONENT_VERSION: ${{ github.event.inputs.component-version }}
  ENV_ID: ${{ github.event.inputs.env-id }}

jobs:
  deploy:
    name: Deploy
    runs-on: ubuntu-latest
    needs: [ prepare ]
    environment: 
      name: ${ENV_ID}
    steps:
[...]
</code></pre>
<p>I want to set <code>jobs.deploy.environment.name</code> with the value of the env var <code>$ENV_ID</code> but instead GitHub creates a new environment named <code>${ENV_ID}</code>.</p>
<p>It look like <code>jobs.deploy.environment.name</code> is NOT supporting any interpolation.</p>
<p>Any idea of how to do that? The only solution I have is to create a workflow for each deployment environment.</p>
<p>Thanks</p>

**Comment by martcus** [comment, marked as answer]
<p>hi!<br>
don’t interpolate the env-id variable.<br>
This works in my test repo:</p>
<pre><code class="lang-auto">name: Enviroment

on:
  workflow_dispatch:
    inputs:
      env-id:
        description: 'Environment ID'
        type: environment
        required: true

jobs:
  deploy:
    environment: 
      name: ${{ github.event.inputs.env-id }}
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3
</code></pre>
<p>i hope this helps you</p>

**Comment by lktslionel** [comment]
<p>Thanks a lot <a class="mention" href="/martcus">@martcus</a>. It works!</p>
